spark/src/test/scala/org/apache/spark/sql/CometTestBase.scala

```diff
@@ -406,6 +408,26 @@ abstract class CometTestBase
     None
   }
 
+  // checks the plan node has no missing inputs
+  // such nodes represented in plan with exclamation mark !
+  // example: !CometWindowExec
+  protected def checkPlanNotMissingInput(plan: SparkPlan): Unit = {
+    def hasMissingInput(node: SparkPlan): Boolean = {
+      node.missingInput.nonEmpty && node.children.nonEmpty
+    }
+
+    val isCometNode = plan.nodeName.startsWith("Comet")
+
+    if (isCometNode && hasMissingInput(plan)) {
+      assert(false, s"Plan node `${plan.nodeName}` has invalid missingInput")
+    }
```

Comment:
```suggestion
    assert(!isCometNode || !hasMissingInput(plan), s"Plan node `${plan.nodeName}` has invalid missingInput")
```

Reply:
Thanks @wForget I'm keeping it that way as other similar assertions 🤷